_drafts/making-backend-scalable.markdown

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+title: "Making Autolab's Backend Scalable"
+author: Ilter Canberk
+
+tags: [tango, redis, scalability]
+
+---
+
+Tango is the back-end service that receives grading jobs from Autolab's front-end, puts them in a job queue, and assigns them to available containers for grading. Tango came across larger work-loads and huge files as Autolab got more popular this semester.
+
```

Comment:
Since this is the first technical post, I think there needs to be a little more background that will establish the problem space. Something like:

```
Tango is a stand-alone RESTful autograding service used by the Autolab app to grade assessments. Tango adds jobs to a job queue, and spawns a worker thread for each job with an available container for grading. In its early days, Tango was mostly used for jobs that ran in under 5 seconds. However, over the recent semesters, Autolab has grown to host classes like Distributed Systems (15-440/640), Machine Learning with Large Datasets (10-605) and Storage Systems (15-746/18-746). Some of these classes have assessments that are computationally intensive on large datasets.
```
```

Also, I think have a link to the Tango API somewhere over here would make sense.